### PR TITLE
Clothing and Armor Coverage Conversion Part 1

### PR DESCRIPTION
**Pull request by Reuvin**
#### Summary
Balance "Converts most of the armor and clothing still using the old limb system to the new system with sub-limbs, part 1: Shoe sole boogaloo, and arm armor too"

#### Purpose of change
There are still a number of armors making use of the old coverage system that uses the entire limb for coverage data, instead of the sub-limb system. This PR is meant to convert all the armors and clothing I can find that would have varied coverages that still use the old system, and then assigning the coverages to the specific sub-limbs instead. 

#### Describe the solution
In part 1, I went through all the arm armor and footwear and converted as much of the coverage values as I felt necessary. If I found an item that needed changing, I would pull up a picture of the item in question and then assign coverages to the sub limbs of the body part it covers in accordance with the real item as best as I could. 

#### Describe alternatives you've considered
Not doing this and saving myself a LOT of time. 

#### Testing
Part 1 is good to go without any errors.

#### Additional context
For the moment this is still in progress. There's a LOT of armor and clothing out there that needs to be converted.

As this is ready to merge as is, I’ll instead do the coverage conversion over a few separate PRs instead of all at once. This part will only contain the arm armor and footwear items.

### CODE REVIEW

**Comment by NetSysFire** [comment]
Whew, thats a lot of work you put into this. A+ for effort.

Now I've got good news and bad news: The bad news is that you need to lint your files. The good news is that this can be done via an automated tool and is basically quick and painless copy pasta. https://dev.narc.ro/cataclysm/format.html

**Comment by Reuvin** [comment]
I was expecting going into this that I'd have to sit around and meticulously fix up one random hidden space at the end of a line I missed or so every so often so the JSON checker would finally stop being mad. 

If the automatic tool works that'll save me a ridiculous amount of time on an already ridiculously time consuming project.

**Comment by NetSysFire** [comment]
>If the automatic tool works

We would have *far* less contributors if we required manual linting and had no tools for this.
But as you can see, you copy pasted correctly and the JSON validator is not mad anymore. Hooray!

You can ignore the failed test, it appears this specific one is broken again.

**Comment by PatrikLundell** [comment]
There's a better way than copy/pasting to/from a web page one file at a time if you've modified many files. tools/format/json_formatter.exe trundles through all the JSON files and changes the ones requiring change.

**Comment by NetSysFire** [comment]
The tool requires manual compiling though, I do not know how advanced this particular contributor is. Seeing that files were added via upload instead of proper git commits suggests they are just getting started. I would only recommend this tool to more advanced contributors who also know how to deal with the commandline. I use said tool a lot and really appreciate it exists since it is way faster than copypasting.

**Comment by Zireael07** [comment]
GitHub upload is absolutely NOT a good tool for such a large project! Things can and likely will break, and you end up with a lot of superfluous commits even if it somehow does not break

File: data/json/items/armor/arms_armor.json [hunk 1]
```diff
@@ -138,12 +138,10 @@
       {
         "covers": [ "arm_l", "arm_r" ],
         "encumbrance": 5,
-        "coverage": 90,
-        "cover_melee": 90,
-        "cover_ranged": 95,
-        "cover_vitals": 90,
```
Comment (bombasticSlacks) [inline data/json/items/armor/arms_armor.json:144]: Can't delete vitals coverage here, it defaults to 0 so having it not defined is bad. 

Also for this one the plastic shouldn't be 100% coverage though the neoprene can be. You can do this by specifying the material coverages. I think 90% covered by plastic is reasonable. 

I'm on mobile rn but could provide an example in a few hours.

**Review by bombasticSlacks — COMMENTED** [review]
Other than my request for the hard arm guards this is mergable as is. 

If you make that change and take this out of draft this can be added and further sublimb armor changes could be made in another PR. 

Also as others have said thank you very much for doing this, its a lot of work and you have done a great job.

File: data/json/items/armor/arms_armor.json (hunk 1 above)
Comment (Reuvin) [reply data/json/items/armor/arms_armor.json:144]: I re-added the vitals coverage as well as specifying varying material coverages for the plastic. Should be good now.

**Comment by Reuvin** [comment]
Just wanted to add this addendum for @bombasticSlacks , when I loaded in to run the last round of testing after touch ups the 'sole' foot part was labelled as the foot arch in game, but it was separate from the actual foot arch and covered the right location on the medical diagram as well as having the proper 'sole' designation in the code. 
image
image
image

**Comment by bombasticSlacks** [comment]
yup that's just a dumb typo on my part, #57933

**Pull request merged**